server/src/main/java/org/elasticsearch/cluster/metadata/MappingMetadata.java

```diff
@@ -88,13 +109,12 @@ public static void writeMappingMetadata(StreamOutput out, ImmutableOpenMap<Strin
         });
     }
 
-    @SuppressWarnings("unchecked")
-    private boolean routingRequired(Map<String, Object> withoutType) {
+    private boolean routingRequired(Map<?, ?> withoutType) {
         boolean required = false;
         if (withoutType.containsKey("_routing")) {
-            Map<String, Object> routingNode = (Map<String, Object>) withoutType.get("_routing");
-            for (Map.Entry<String, Object> entry : routingNode.entrySet()) {
-                String fieldName = entry.getKey();
+            Map<?, ?> routingNode = (Map<?, ?>) withoutType.get("_routing");
```

Comment:
Why can't we have `String` keys here anymore?

Reply:
I was a bit annoyed with the `SuppressWarnings("unchecked")` and so removed it. I can undo this change and make this change in another pr.

Reply:
Maybe, this one is a little too impactful to have additional noise in it I think :)

Reply:
Ok, I will undo this bit here in the pr.

Reply:
Removed this change: de25c023076ed9d533cd785f60bdeaad3bbd4368